### DISCUSSION TITLE
Plotfiles vs. Checkpoint files

### DISCUSSION THREAD

**Discussion by RadekPecher** [AMReX-Codes/amrex#4185] [Q&A]
I would like to understand why AMReX supports two grid/particle output formats, one devoted to plotting (plotfiles) while the other to restart runs (checkpoint files). It seems a bit wasteful, hence I plan to use just one format for both plotting and restarts. But when I tried that, there were issues. Plotfiles appear to be partially supporting restarts (see e.g. `Amr::restart_pltfile`) but I assume they might be missing some data required for restarts (right?). Whereas checkpoint files can't be imported into ParaView, which I'm using for my results visualisation. Could you please explain? Many thanks.

**Comment by zingale** [comment]
My take on why they are different: they serve different purposes and we often want to output them at different frequencies and with different data.

Plotfiles can have derived data in them, which we make use of a lot, to compute some state on the grid that isn't carried by the code.  We typically also output plotfiles at fixed simulation time intervals (since we want to make smooth animations) and output checkpoint files at fixed step intervals (since we want to ensure that we output a checkpoint in a queue window when running on a supercomputer).

**Reply by drummerdoc** [reply]
Agreed.  Lots of history there as well.  Another difference is that checkpoints can have MultiFab data with all different centerings as needed for different types of solvers.  While plotfiles *could* have similar, only cell-centered data is supported and read by the few viz packages that understand them.  Support for all the centerings would require considerably more complex infrastructure that doesn't yet exist.  Also, since plotfiles are tuned by the user they can be as light- or heavy-weight as needed for an application, whereas checkpoints contain exactly everything needed to continue a run as before (nothing more and nothing less).  

All that said, each can be switched off individually during a run, and you can build your own analysis tools to expect whatever you might need.  For example, in Pele, we often restart from plotfiles when making a significant change to the run (changing the chemistry model, add reactions/radiation/soot or Lagrangian sprays, where restarting from a checkpoint would require that all the requisite data structures be available at restart rather than being initialized (but perhaps we want to start from a developed velocity field).

Some of these historical things can be frustrating, but most have useful rationales and decent workarounds.  If you need help creating other workarounds, post here.  Hope that helps.
